### PR TITLE
Add quadratic form

### PR DESCRIPTION
**Pull request by toor1245**

### CODE REVIEW

File: src/MatrixDotNet/Extensions/QudraticForm/Form.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+namespace MatrixDotNet.Extensions.QudraticForm
+{
+    /// <summary>
+    /// Represents quadratic form.
```
Comment (Dilorfin) [inline src/MatrixDotNet/Extensions/QudraticForm/Form.cs:4]: But wait, the quadratic form is a polynomial...

image


It seems to me that you meant '[type of definite symmetric matrix](https://en.wikipedia.org/wiki/Definite_symmetric_matrix)'

**Review by Dilorfin — CHANGES_REQUESTED** [review]

**Comment by Dilorfin** [comment]
For #21 "src/MatrixDotNet.NetCore/MatrixAsFixedBuffer.cs#L314" should be refactored

**Comment by toor1245** [comment]
> For #21 "src/MatrixDotNet.NetCore/MatrixAsFixedBuffer.cs#L314" should be refactored

Ok, I will change it.

File: src/MatrixDotNet.sln.DotSettings [hunk 2]
```diff
@@ -2,15 +2,19 @@
...
 	<s:Boolean x:Key="/Default/UserDictionary/Words/=Convertor/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/UserDictionary/Words/=Criteries/@EntryIndexedValue">True</s:Boolean>
+	
 	<s:Boolean x:Key="/Default/UserDictionary/Words/=Frobenius/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/UserDictionary/Words/=Gramm/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/UserDictionary/Words/=intelmkl/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/UserDictionary/Words/=Kahan/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/UserDictionary/Words/=Klain/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/UserDictionary/Words/=Pinnable/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/UserDictionary/Words/=Qudratic/@EntryIndexedValue">True</s:Boolean>
```
Comment (Dilorfin) [inline src/MatrixDotNet.sln.DotSettings:13]: It seems to me that this word should be removed, at least for now

Comment (toor1245) [reply src/MatrixDotNet.sln.DotSettings:13]: > It seems to me that this word should be removed, at least for now

I totally agree with you.

**Comment by toor1245** [comment]
> For #21 "src/MatrixDotNet.NetCore/MatrixAsFixedBuffer.cs#L300" should be refactored

I just changed

**Comment by toor1245** [comment]
@Dilorfin, maybe you can add a test for macOS too?

File: src/MatrixDotNet/Matrix.cs [hunk 3]
```diff
@@ -66,7 +66,12 @@ public bool IsPrime
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             get => (Rows & 0b01) == 0 && (Columns & 0b01) == 0;
         }
-        
+
+        /// <summary>
+        /// Checks for symmetric matrix.
+        /// </summary>
+        public bool IsSymmetric => this == this.Transpose();
```
Comment (Dilorfin) [inline src/MatrixDotNet/Matrix.cs:73]: OK for now, but open #32

**Review by Dilorfin — DISMISSED** [review]

**Comment by Dilorfin** [comment]
> @Dilorfin, maybe you can add a test for macOS too?

Ok

**Review by Dilorfin — APPROVED** [review]

**Pull request merged**